packages/job-dispatch/src/job-variables-deployment/job-variables-deployment.ts

```
@@ -0,0 +1,149 @@
+        return;
+      }
+
+      if (directMatches.includes(val.key)) return;
+
+      const existingVariableIdx = jobVariables.findIndex(
+        (v) => v.key === val.key,
+      );
+
+      if (existingVariableIdx === -1) return;
+
+      jobVariables[existingVariableIdx]!.value = val.value;
```

Biome

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)